corehq/apps/export/models/new.py

```diff
@@ -0,0 +1,342 @@
+    this question.
+    """
+    options = SchemaListProperty(Option)
+
+    @classmethod
+    def create_from_question(cls, question, appVersion):
+        item = super(MultipleChoiceItem, cls).create_from_question(question, appVersion)
+
+        for option in question.options:
+            item.options.append(Option(
+                last_occurrence=appVersion,
+                value=option['value']
+            ))
+        return item
+
+    @classmethod
+    def merge(cls, one, two):
+        item = super(MultipleChoiceItem, cls).merge(one, two)
+        options = _merge_lists(one.options, two.options,
+            keyfn=lambda i: i.value,
+            resolvefn=lambda option1, option2:
+                Option(
+                    value=option1.value,
+                    last_occurrence=max(option1.last_occurrence, option2.last_occurrence)
+                ),
+            copyfn=lambda option: Option(option.to_json())
+        )
+
+        item.options = options
+        return item
+
+
+class ExportGroupSchema(DocumentSchema):
+    """
+    An object that represents a logical group of questions in the form
+    """
+    path = ListProperty()
+    items = SchemaListProperty(ExportItem)
+    last_occurrence = IntegerProperty()
+
+
+class ExportDataSchema(DocumentSchema):
+    """
+    An object representing the things that can be exported for a particular
+    form xmlns or case type.
+    Each item in the list is uniquely identified by its path and doc_type.
+    repeats is a list of RepeatGroups, present at any level of the question hierarchy
+    """
+    group_schemas = SchemaListProperty(ExportGroupSchema)
+    datatype_mapping = defaultdict(lambda: ScalarItem, {
+        'MSelect': MultipleChoiceItem,
+    })
+
+    @staticmethod
+    def generate_schema_from_builds(domain, app_id, unique_form_id):
+        app_build_ids = get_built_app_ids_for_app_id(domain, app_id)
+        all_xform_conf = TableConfiguration()
+
+        for app in iter_docs(Application.get_db(), app_build_ids):
+            xform = app.get_form(unique_form_id).wrapped_xform()
+            xform_conf = ExportDataSchema._generate_schema_from_xform(xform, app.langs, app.version)
+            all_xform_conf = ExportDataSchema._merge_schema(all_xform_conf, xform_conf)
+
+        return all_xform_conf
+
+    @staticmethod
+    def _generate_schema_from_xform(xform, langs, appVersion):
+        questions = xform.get_questions(langs)
+        schema = ExportDataSchema()
+
+        for group_path, group_questions in groupby(questions, lambda q: q['repeat']):
+            # If group_path is None, that means the questions are part of the form and not a repeat group
+            # inside of the form
+            group_schema = ExportGroupSchema(
+                path=_string_path_to_list(group_path),
+                last_occurrence=appVersion,
+            )
+            for question in group_questions:
+                wrapped_question = Question(**question)
```

Comment:
not necessary. it was more for readability. i kept having to go into `get_questions` function to see what was defined on the question since the question is just a dict. think we can remove if it's superfluous